File: wesignmanagement-client-DEV/src/app/models/report-parameters.ts
import { SignatureType } from "./signature-type.enum";

export class ReportParameters {
    public offset: number;
    public limit: number;
    public programUtilizationHistoryKey: string;
    public isProgramUtilizationExpired: boolean;
    public monthsForAvgUse: number;
    public programId: string;
    public docsUsagePercentage: number;
    public companyId: string;
    public groupIds: string[];
    public minDocs: number;
    public minSms: number;
    public isProgramUsed: boolean;
    public userEmail: string;
    public signatureTypes: SignatureType[];
}

File: user-backend/Common/Models/ManagementApp/Reports/ReportParameters.cs
using Common.Enums.PDF;
using System;
using System.Collections.Generic;

namespace Common.Models.ManagementApp.Reports
{
    public class ReportParameters
    {
        public int Offset { get; set; } // Global
        public int Limit { get; set; } // Global
        public string ProgramUtilizationHistoryKey { get; set; } // 1
        public bool? IsProgramUtilizationExpired { get; set; } // 1
        public int MonthsForAvgUse { get; set; } // 5
        public Guid ProgramId { get; set; } // 2
        public int DocsUsagePercentage { get; set; } // 2
        public Guid CompanyId { get; set; } // 9
        public List<Guid> GroupIds { get; set; } // 5
        public int MinDocs { get; set; } // 1
        public int MinSms { get; set; } // 1
        public bool? IsProgramUsed { get; set; } // 1
        public string UserEmail { get; set; } // 1
    }
}
